python/lsst/daf/butler/script/pruneCollection.py

```diff
@@ -42,14 +59,81 @@ def pruneCollection(repo, collection, purge, unstore, unlink):
         Same as the ``unstore`` argument to ``Butler.pruneCollection``.
     unlink: `list` [`str`]
         Same as the ``unlink`` argument to ``Butler.pruneCollection``.
+    confirm : `bool`
+        If `True` will produce a table of collections that will be removed for
+        display to the user.
+
+    Returns
+    -------
+    collections : `astropy.table.Table`
+        The table containing collections that will be removed, their type, and
+        the number of datasets in the collection if applicable.
     """
-    butler = Butler(repo, writeable=True)
-    try:
-        butler.pruneCollection(collection, purge, unstore, unlink)
-    except PurgeWithoutUnstorePruneCollectionsError as e:
-        raise TypeError("Cannot pass --purge without --unstore.") from e
-    except RunWithoutPurgePruneCollectionsError as e:
-        raise TypeError(f"Cannot prune RUN collection {e.collectionType.name} without --purge.") from e
-    except PurgeUnsupportedPruneCollectionsError as e:
-        raise TypeError(
-            f"Cannot prune {e.collectionType} collection {e.collectionType.name} with --purge.") from e
+    class CollectionInfo:
+        def __init__(self, count, type=None):
```

Comment:
Can you please put a little bit of documentation in this class? In particular, type annotations. Note that if you turn mypy on for this file it really isn't going to be happy about `self.count` being an int or a string when code later on assumes that it will be just fine to increment it. Given that it's a private class you can get away with it but you'd probably want to write this differently once we go to mypy everywhere (in some future dream land).

Reply:
Added docs & type annotations. Let me know if it looks ok? During development I temporarily turned off the `disallow_untyped_defs` exception for the scripts folder, and made sure there were no reported untyped defs for this file, so hopefully this file is totally taken care of now.